### ISSUE TITLE
Additional methods for Bhom_Engine

### ISSUE THREAD

**Issue by vietle-bh** [BHoM/BHoM_Engine#3034]
#### Description:

We need a few additional methods to handle new use cases such as:
- Converting between degree and radian values
- Copy property values between 2 objects where the property names match
- Shifting and splitting a list

**Comment by FraserGreenroyd** [comment]
The first method already exists within the localisation toolkit and would not be suitable for BHoM engine as the engine should stay in SI.

The second method I thought already exists but it'll be in the reflection engine. If it doesn't exist, then that would be the better engine for it I think.

Third method, worth checking in both data engine and environment engine (@tg359 you might have done something?) as again I think they might already exist but in there. If they do exist in the environment engine then they should be moved up to a more general engine.

**Comment by vietle-bh** [comment]
Thanks @FraserGreenroyd , I'll make sure the check these as you said.
Happy Easter!! 🐰

**Comment by vietle-bh** [comment]
Hi @FraserGreenroyd , I made the following changes:
- Removed my conversion methods between degree and radian
- Moved my method for copying properties to Reflection Engine. There's another property copying method there but it requires the 2 objects to be of parent and child classes. My method doesn't have this requirement as it matches properties by name.
- Methods for shifting and splitting a list: I tried but haven't found anything similar in data engine and environment engine. For now, my methods are still here:
  - https://github.com/BHoM/BHoM_Engine/blob/a18bc531708ac4af48b05a94389dd4680abbf682/BHoM_Engine/Modify/SplitAndRemoveAtIndexes.cs
  - https://github.com/BHoM/BHoM_Engine/blob/bcf7568e5d54ce6aca3728e4be768172d90fbafc/BHoM_Engine/Modify/ShiftList.cs
